src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs

```diff
@@ -2052,6 +2052,93 @@ partial class B<U, V> where U : T where V : U, U, IB<T>, IB<T> { }
                 Diagnostic(ErrorCode.ERR_DuplicateBound, "IB<T>").WithArguments("C<T>.IB<T>", "V").WithLocation(14, 62));
         }
 
+        [Fact]
+        public void DuplicateConstraintDifferencesOnPartialMethod()
+        {
+            var source =
+@"interface I<T> { }
+partial class C<T>
+{
+    partial void F<U>() where U : T, T, I<T>;
+    partial void F<U>() where U : T, I<T>, I<T> { }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (4,38): error CS0405: Duplicate constraint 'T' for type parameter 'U'
+                //     partial void F<U>() where U : T, T, I<T>;
+                Diagnostic(ErrorCode.ERR_DuplicateBound, "T").WithArguments("T", "U").WithLocation(4, 38),
+                // (5,44): error CS0405: Duplicate constraint 'I<T>' for type parameter 'U'
+                //     partial void F<U>() where U : T, I<T>, I<T> { }
+                Diagnostic(ErrorCode.ERR_DuplicateBound, "I<T>").WithArguments("I<T>", "U").WithLocation(5, 44));
+        }
+
+        [Fact]
+        public void ConstraintErrorMultiplePartialDeclarations_01()
```

Comment:
>ConstraintErrorMultiplePartialDeclarations_01 [](start = 20, length = 45)

Maybe test partial declarations with differences in `?`. For instance, one with an `I` constraint, and the other with an `I?` constraint.

Reply:
See `StaticNullChecking.PartialClassConstraintMismatch`.

---
In reply to: [197928471](https://github.com/dotnet/roslyn/pull/27525#discussion_r197928471) [](ancestors = 197928471)